engine/src/player-platform.cpp

```diff
@@ -2119,7 +2122,7 @@ Boolean MCPlayer::prepare(const char *options)
    	if (!opened)
 		return False;
     
-	if (m_platform_player == nil)
+	if (m_platform_player == nil || m_should_recreate)
```

Comment:
Won't this cause a memory leak if m_should_recreate is true, and m_platform_player != nil?